### PR TITLE
hello world java with tabs

### PR DESCRIPTION
**Pull request by itsmurugappan**
## Proposed Changes 

- hello world java example with tabs

### CODE REVIEW

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (mpetason) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:136]:
```suggestion
1. After the build has completed and the container is pushed to Docker Hub, you
```

Capitalized for consistency

**Review by maximilien — APPROVED** [review]
LGTM

**Comment by tcnghia** [comment]
```

=== RUN   TestDocSrc/java-spring     sampleconsistency_test.go:97: README.md file is missing line 0 ('package com.example.helloworld;') from file '../../docs/serving/samples/hello-world/helloworld-java-spring/src/main/java/com/example/helloworld/HelloworldApplication.java'         Additional info:         All files required for running sample apps are checked against README.md, the content of source files should be identical with what's in README.md file, the list of the files to be verified is the same set of files used for running sample apps, they are configured in `/test/sampleapp/config.yaml`. If an exception is needed the file can be configured to be copied as a separate step in `PreCommand` such as: https://github.com/knative/docs/blob/65f7b402fee7f94dfbd9e4512ef3beed7b85de66/test/sampleapp/config.yaml#L4     --- FAIL: TestDocSrc/java-spring (0.00s) | === RUN   TestDocSrc/java-spring     sampleconsistency_test.go:97: README.md file is missing line 0 ('package com.example.helloworld;') from file '../../docs/serving/samples/hello-world/helloworld-java-spring/src/main/java/com/example/helloworld/HelloworldApplication.java'         Additional info:         All files required for running sample apps are checked against README.md, the content of source files should be identical with what's in README.md file, the list of the files to be verified is the same set of files used for running sample apps, they are configured in `/test/sampleapp/config.yaml`. If an exception is needed the file can be configured to be copied as a separate step in `PreCommand` such as: https://github.com/knative/docs/blob/65f7b402fee7f94dfbd9e4512ef3beed7b85de66/test/sampleapp/config.yaml#L4     --- FAIL: TestDocSrc/java-spring (0.00s)
-- | --
=== RUN   TestDocSrc/java-spring     sampleconsistency_test.go:97: README.md file is missing line 0 ('package com.example.helloworld;') from file '../../docs/serving/samples/hello-world/helloworld-java-spring/src/main/java/com/example/helloworld/HelloworldApplication.java'         Additional info:         All files required for running sample apps are checked against README.md, the content of source files should be identical with what's in README.md file, the list of the files to be verified is the same set of files used for running sample apps, they are configured in `/test/sampleapp/config.yaml`. If an exception is needed the file can be configured to be copied as a separate step in `PreCommand` such as: https://github.com/knative/docs/blob/65f7b402fee7f94dfbd9e4512ef3beed7b85de66/test/sampleapp/config.yaml#L4     --- FAIL: TestDocSrc/java-spring (0.00s)
 


```

**Comment by itsmurugappan** [comment]
[pr](https://github.com/knative/docs/pull/2832) takes care of this test failure, once that is merged , i can rebase which will fix the test

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:10]:
```suggestion
The sample app reads a `TARGET` environment variable, and prints `Hello ${TARGET}!`.
If `TARGET` is not specified, `World` is used as the default value.
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:25]:
```suggestion
- Optional: the Knative CLI client [kn](https://github.com/knative/client/releases) can be used to simplify deployment. Alternatively, you can use `kubectl`, and apply resource files directly.
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:27]:
```suggestion
## Building the sample app
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:30]:
```suggestion
1. From the console, create a new, empty web project by using the curl and unzip
   commands:
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:41]:
```suggestion
   If you don't have `curl` installed, you can accomplish the same by visiting the
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:49]:
```suggestion
   provide the `TARGET` environment variable:
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:93]:
```suggestion
1. In your project directory, create a file named `Dockerfile` and copy the code
   block below into it:
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:121]:
```suggestion
For detailed instructions on dockerizing a Spring Boot app, see [Spring Boot with Docker](https://spring.io/guides/gs/spring-boot-docker/).

For additional information on multi-stage docker builds for Java, see [Creating Smaller Java Image using Docker Multi-stage Build](http://blog.arungupta.me/smaller-java-image-docker-multi-stage-build/).
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:120]: No need for this line, the title of the section already tells us we're building an app. Removed lines.

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [reply docs/serving/samples/hello-world/helloworld-java-spring/index.md:121]: Moved these links until the end of the procedure. Generally good practice to only have steps in a procedure and any links, refs etc go in an Additional Resources section or similar at the end.

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:125]:
```suggestion
**NOTE:** Use Docker to build the sample code into a container. To build and push with Docker Hub, run these commands replacing `{username}` with your Docker Hub username.

1. Build the sample code into a container, then push the container to the Docker registry:
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:136]:
```suggestion
After the build has completed and the container is pushed to Docker Hub, you
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:134]:
```suggestion
## Deploying the app
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:189]:
```suggestion
   - Automatically scale your pods up and down, including scaling down to zero active pods.
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:193]:
```suggestion
1. Find the domain URL for your service:
```

File: docs/serving/samples/hello-world/helloworld-java-spring/index.md
Comment (abrennan89) [inline docs/serving/samples/hello-world/helloworld-java-spring/index.md:224]:
```suggestion
1. Make a request to your app and observe the result. Replace
   the URL below with the URL returned in the previous command.
```

**Review by abrennan89 — CHANGES_REQUESTED** [review]
Some minor comments

**Comment by abrennan89** [comment]
> [pr](https://github.com/knative/docs/pull/2832) takes care of this test failure, once that is merged , i can rebase which will fix the test

@itsmurugappan Thanks for the updates! Do we have a dependency on merging #2832 first, or is this PR ready to merge now?
@rhuss can you take a look at that PR please, it looks like it needs a rebase.

**Review by abrennan89 — APPROVED** [review]
/hold
/approve
/lgtm

**Comment by itsmurugappan** [comment]
This can be merged now as I have fixed the Readme to pass the tests.

**Comment by abrennan89** [comment]
Thanks @itsmurugappan - does this need to be cherry-picked to any older branches?

**Pull request merged**
